unit/java_bytecode/java_bytecode_parse_generics/parse_generic_superclasses.cpp

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************\
+
+ Module: Unit tests for parsing classes with generic superclasses with
+         unsupported signatures, falling back to using the raw type
+         descriptors
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+SCENARIO(
+  "parse generic superclass signature",
+  "[core][java_byte code[java_bytecode_parse_generics]]")
+{
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "GenericBounds", "./java_bytecode/java_bytecode_parse_generics");
+
+  const std::string load_class("java::GenericBounds");
+  REQUIRE(new_symbol_table.has_symbol(load_class));
+  THEN("symbol is of non-generic class type")
+  {
+    const symbolt &class_symbol = new_symbol_table.lookup_ref(load_class);
+
+    const typet &class_type = class_symbol.type;
+    REQUIRE(class_type.id() == ID_struct);
+
+    const struct_typet &class_struct = to_struct_type(class_type);
```

Comment:
Shouldn't this test be checking the bases of the symbol rather than the components?

Reply:
I agree, this should be checking bases - you can use functions `require_symbol` and `require_java_generic_symbol_type` from `testing-utils/require_type.cpp`. There are similar tests in `parse_derived_generic_class.cpp` so consider extending that file (and the corresponding java file) instead of creating a new test.

Reply:
This still seems to be outstanding